app/Http/Controllers/ProjectController.php

```diff
@@ -2,30 +2,152 @@
 
 namespace App\Http\Controllers;
 
+use Illuminate\Http\Request;
+use App\Models\Project;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\Input;
+use Illuminate\Support\Facades\Session;
+
+
 class ProjectController extends Controller
 {
     /**
-     * Create new project
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        // get all the projects
+        $projects = Project::all();
+
+        return view('pages.projects.index', compact('projects'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
      */
     public function create()
     {
+        return view('pages.projects.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        // validate
+        // read more on validation at http://laravel.com/docs/validation
+        $rules = array(
+            'name'              => 'required',
+            'repository_url'    => 'required'
+        );
+        $validator = Validator::make(Input::all(), $rules);
```

Comment:
Take a look at https://laravel.com/docs/5.6/validation#form-request-validation. This will remove lines 46-58